tests/src/system/basic/Swift3WhiskObjectTests.scala

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2015-2016 IBM Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package system.basic
+
+import org.junit.runner.RunWith
+import org.scalatest.Finders
+import org.scalatest.junit.JUnitRunner
+
+import common.TestHelpers
+import common.TestUtils
+import common.Wsk
+import common.WskProps
+import common.WskTestHelpers
+import spray.json.DefaultJsonProtocol.StringJsonFormat
+import spray.json.pimpAny
+
+@RunWith(classOf[JUnitRunner])
+class Swift3WhiskObjectTests
+    extends TestHelpers
+    with WskTestHelpers {
+
+    implicit val wskprops = WskProps()
+    val wsk = new Wsk()
+
+    behavior of "Swift 3 Whisk backend API"
+
+    it should "allow Swift actions to invoke other actions" in withAssetCleaner(wskprops) {
+        (wp, assetHelper) =>
+            // use Wsk to create action from dat/actions/invokeAction.swift
+            val file = TestUtils.getTestActionFilename("invokeAction.swift")
+            val actionName = "invokeAction"
+            assetHelper.withCleaner(wsk.action, actionName) { (action, _) =>
+                action.create(name = actionName, artifact = Some(file), kind = Some("swift:3"), parameters = Map("key0" -> "value0".toJson), expectedExitCode =0)
+            }
+
+            // invoke the action
+            val invokeResult = wsk.action.invoke(actionName, Map("key0" -> "value0".toJson), blocking = true, result = false, 0).stdout
```

Comment:
should turn this into a non-blocking invoke and use the new `withActivation` idiom.